client/src/components/overview/ImageNavigatorModel.js

```diff
@@ -0,0 +1,138 @@
+class ImageNavigatorModel {
+  constructor({
+    thumbnails = [],
+    useIcons = false,
+    length = 0,
+    maxItems = 7,
+    startIndex = 0
+  } = {}) {
+    this.thumbnails = thumbnails ? thumbnails : [];
+    this.useIcons = useIcons;
+    this.startIndex = startIndex;
+    this.maxItems = maxItems;
+
+    this.length = this._getLength(thumbnails, maxItems, length);
+    this.endIndex = this._getEndIndex(startIndex, maxItems, this.length);
+  }
+
+  _getEndIndex(startIndex, maxItems, length) {
+    return Math.min(startIndex + maxItems - 1, length - 1);
+  }
+
+  _getLength(thumbnails, maxItems, length) {
+    if (thumbnails && thumbnails.length > 0) {
+      return thumbnails.length;
+    } else if (length) {
+      return length;
+    } else {
+      return maxItems;
+    }
+  }
+
+  usePlaceholders() {
```

Comment:
Maybe consider styling the `<img>` element or its container so that it defaults to a placeholder whenever `src` is an empty string.